parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeospatialStatistics.java

```diff
@@ -0,0 +1,250 @@
+
+import org.apache.parquet.Preconditions;
+import org.apache.parquet.column.schema.EdgeInterpolationAlgorithm;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A structure for capturing metadata for estimating the unencoded,
+ * uncompressed size of geospatial data written.
+ */
+public class GeospatialStatistics {
+  private static final Logger LOG = LoggerFactory.getLogger(GeospatialStatistics.class);
+
+  // Metadata that may impact the statistics calculation
+  private final BoundingBox boundingBox;
+  private final EdgeInterpolationAlgorithm edgeAlgorithm;
+  private final GeospatialTypes geospatialTypes;
+
+  /**
+   * Merge the statistics from another GeospatialStatistics object.
+   *
+   * @param other the other GeospatialStatistics object
+   */
+  public void mergeStatistics(GeospatialStatistics other) {
+    if (other == null) {
+      return;
+    }
+    if (this.boundingBox != null && other.boundingBox != null) {
+      this.boundingBox.merge(other.boundingBox);
+    }
+    if (this.geospatialTypes != null && other.geospatialTypes != null) {
+      this.geospatialTypes.merge(other.geospatialTypes);
+    }
+  }
+
+  /**
+   * Builder to create a GeospatialStatistics.
+   */
+  public static class Builder {
+    private BoundingBox boundingBox;
+    private GeospatialTypes geospatialTypes;
+    private EdgeInterpolationAlgorithm edgeAlgorithm;
+    private final WKBReader reader = new WKBReader();
+
+    /**
+     * Create a builder to create a GeospatialStatistics.
+     * For Geometry type, edgeAlgorithm is not required.
+     */
+    public Builder() {
+      this.boundingBox = new BoundingBox();
+      this.geospatialTypes = new GeospatialTypes();
+      this.edgeAlgorithm = null;
+    }
+
+    /**
+     * Create a builder to create a GeospatialStatistics.
+     * For Geography type, optional edgeAlgorithm can be set.
+     */
+    public Builder(EdgeInterpolationAlgorithm edgeAlgorithm) {
+      this.boundingBox = new BoundingBox();
+      this.geospatialTypes = new GeospatialTypes();
+      this.edgeAlgorithm = edgeAlgorithm;
+    }
+
+    public void update(Binary value) {
+      if (value == null) {
+        return;
+      }
+      try {
+        Geometry geom = reader.read(value.getBytes());
+        update(geom);
+      } catch (ParseException e) {
+        LOG.warn("Failed to parse WKB geometry, aborting statistics update", e);
+        abort();
+      }
+    }
+
+    private void update(Geometry geom) {
+      boundingBox.update(geom);
+      geospatialTypes.update(geom);
+    }
+
+    public void abort() {
+      boundingBox.abort();
+      geospatialTypes.abort();
+    }
```

Comment:
`geospatialTypes` has a `valid` field, it will be set to `false` once we abort the update, no further written geometries can revive it from invalid state to valid state.

However, `boundingBox` does not have a flag indicating if are in invalid state because of aborting. It could revive to valid state if we write valid geometries following up the invalid geometry. This could be dangerous and the incorrect bounding box may appear in the geometry statistics of the parquet file being written.

Can we handle the invalid state of `boundingBox` and `geospatialTypes` in a consistent way? Can we add a test case to show that geospatial statistics won't appear in the written parquet file when invalid geometries were mixed into the values being written?